### PR TITLE
feature/brc20

### PR DESCRIPTION
**Pull request by StewartYe**

### CODE REVIEW

File: customs/brc20/src/bitcoin_to_custom.rs [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+                );
+            }
+            Some(token) => {
+                let args = create_query_brc20_transfer_args(
+                    gen_ticket_request.clone(),
+                    deposit_addr.clone(),
+                    token.decimals,
+                );
+                let query = query_indexed_transfer(args).await;
+                if let Ok(Some(_t)) = query {
```
[the remark this answers is not in the document]
Comment (StewartYe) [reply customs/brc20/src/bitcoin_to_custom.rs:160]: checked in brc20_indexer

File: customs/brc20/src/service.rs [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+    init_ecdsa_public_key().await;
+    read_state(|s| (s.deposit_addr.clone(), s.deposit_pubkey.clone()))
+}
+
+#[query(guard = "is_admin")]
+pub fn brc20_state() -> StateProfile {
+    read_state(|s| StateProfile::from(s))
+}
+
+#[update]
```
[the remark this answers is not in the document]
Comment (StewartYe) [reply customs/brc20/src/service.rs:61]: any for test ,will process when lauch to prod

**Review by boern — APPROVED** [review]

**Review by hsxyl — APPROVED** [review]

File: customs/brc20/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+        T,
+        E: Default + Clone + ToString + Debug,
+        R: Future<Output = Result<T, E>>,
+    >(
+        params: P,
+        call_rpc: fn(params: P) -> R,
+    ) -> Result<T, E> {
+        let mut rs = Err(E::default());
+        for i in 0..RPC_RETRY_TIMES {
+            log!(INFO, "[evm route]request rpc request times: {}", i + 1);
```
Comment (en) [inline customs/brc20/src/lib.rs:72]: evm route -> brc20 customs

File: customs/brc20/src/lib.rs [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+            let call_res = call_rpc(params.clone()).await;
+            if call_res.is_ok() {
+                rs = call_res;
+                break;
+            } else {
+                let err = call_res.err().unwrap();
+                log!(
+                    ERROR,
+                    "[evm route]call  rpc error: {}",
```
Comment (en) [inline customs/brc20/src/lib.rs:81]: evm route -> brc20 customs

**Pull request merged**